prow/flagutil/github.go

```diff
@@ -47,25 +47,87 @@ type GitHubOptions struct {
 	AppID             string
 	AppPrivateKeyPath string
 
+	ThrottleHourlyTokens int
+	ThrottleAllowBurst   int
+
 	// This will only be set after a github client was retrieved for the first time
 	appsTokenGenerator github.GitHubAppTokenGenerator
 }
 
+// flagParams struct is used indirectly by users of this package to customize
+// the common flags behavior, such as providing their own default values
+// or suppressing presence of certain flags.
+type flagParams struct {
+	defaults GitHubOptions
+
+	denyThrottlerOptions bool
+}
+
 const DefaultGitHubTokenPath = "/etc/github/oauth" // Exported for testing purposes
 
-// AddFlags injects GitHub options into the given FlagSet.
+type FlagParameter func(options *flagParams)
+
+// ThrottlerDefaults allows to customize the default values of flags
+// that control the throttler behavior. Setting `hourlyTokens` to zero
+// disables throttling by default.
+func ThrottlerDefaults(hourlyTokens, allowedBursts int) FlagParameter {
+	return func(o *flagParams) {
+		o.defaults.ThrottleHourlyTokens = hourlyTokens
+		o.defaults.ThrottleAllowBurst = allowedBursts
+	}
+}
+
+// DenyThrottlerOptions suppresses the presence of throttler-related flags,
+// effectively disallowing external users to parametrize default throttling
+// behavior. This is useful mostly when a program creates multiple GH clients
+// with different behavior.
+func DenyThrottlerOptions() FlagParameter {
```

Comment:
```suggestion
func DisableThrottlerOptions() FlagParameter {
```

Reply:
#drunkcoding but I do not drink anymore so /shrug

Reply:
Fixed everywhere.